src/hooks/useDetectNetworkConnection.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  useCallback,
+  createContext,
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { useNetInfo } from '@react-native-community/netinfo';
+import { t } from 'i18next';
+import Toast, { ToastProps } from 'react-native-root-toast';
+
+const DetectNetworkConnectionContext = createContext<{
+  isConnected: boolean | null;
+  checkConnection?: () => void;
+}>({
+  isConnected: true,
+});
+
+export const DetectNetworkConnectionProvider = ({
+  children,
+  toastOverrides,
+}: {
+  children?: React.ReactNode;
+  toastOverrides?: ToastProps;
+}) => {
+  const { isConnected } = useNetInfo();
+  const [forceCheck, setForceCheck] = useState<boolean>(false);
+  const toastRef = useRef(undefined);
```

Comment:
If I'm understanding the PR, this is combining the hook that's used if consumers want an `isConnected` prop to a user display (the toast).  I think ideally those should stay separate.  How would you feel about separating those two into a focused hook (possibly just migrating `useHasConnection`) and then a separate toast display component which utilizes its prop(s)?

Reply:
Since the toast is already its own component and has an API and netinfo already provides a hook for getting the connection state I am not sure what those hooks would provide. It wasn't my intention to use this like `useHasConnection` hook, i.e., that is a hook wrapper around an existing hook that doesn't really extend its functionality. In my mind developer's would just use netinfo directly for their connection related needs BUT I also exposed it here for a convenience but perhaps that just muddies the intention of this.

My intent was this would be a simple "detect network outage toast" with a way to force a recheck, not the go to hook for all network connection related matters. 

Does that all make sense? If so I think at least a rename is probably in order.

Reply:
Ah, yes that totally makes sense.  I think you're spot on that the exposure of `checkConnection` and `isConnected` is what got us off on the wrong track.  I agree with @sternetj [here](https://github.com/lifeomic/react-native-sdk/pull/60/files#r1120169919) too so removing `checkConnection` exposure altogether would resolve that conversation as well.

Reply:
Yeah that definitely clears it up for me and makes perfect sense.